src/index.ts

```diff
@@ -122,6 +122,28 @@ const isPromise = <T>(value: unknown): value is Promise<T> =>
   (typeof value === "object" || typeof value === "function") &&
   typeof (value as { then?: Function }).then === "function";
 
+export const createOptionalValidator =
+  <Value, ErrorMessage = string>(
+    validator: Validator<Value, ErrorMessage>,
+    emptyValue?: Value,
+  ): Validator<Value, ErrorMessage> =>
+  (value) => {
+    const isEmptyValue = value === (typeof emptyValue !== "undefined" ? emptyValue : "");
+    const result = validator(value);
```

Comment:
maybe let's not run it in case `isEmptyValue` is true, could have side-effects or expensive computations

Reply:
Ah yes, I'm stupid. Could be much more simpler.